server/embed/doc.go

```diff
@@ -31,13 +31,15 @@ Launch an embedded etcd server using the configuration defaults:
 		if err != nil {
 			log.Fatal(err)
 		}
+
+		if e.Server != nil {
+			e.Server.Stop()
+		}
+
 		defer e.Close()
 		select {
 		case <-e.Server.ReadyNotify():
 			log.Printf("Server is ready!")
-		case <-time.After(60 * time.Second):
-			e.Server.Stop() // trigger a shutdown
-			log.Printf("Server took too long to start!")
 		}
 		log.Fatal(<-e.Err())
```

Comment:
I think the program will be always blocked on this line?

Reply:
Please follow https://github.com/etcd-io/etcd/discussions/15724#discussioncomment-5667005 to update the doc, thx

Reply:
Just in case if it's not clear, I guess what @ahrtr meant is using the same logic in

1. start embed etcd server. 
2. register `e.Close` into interrupt handler. 
https://github.com/etcd-io/etcd/blob/372042c374ceea799e52c406b21a01ee2c5bab49/server/etcdmain/etcd.go#L205-L218

3. listen on SIGINT and SIGTERM to process `e.Close`.
4. process server initialization async failures or stopped signals.

Reply:
thx @chaochn47 for the clarification.